src/test/resources/projects/standard-project/pom.xml

```diff
@@ -14,30 +14,79 @@
   <packaging>war</packaging>
   <name>App Engine Standard Project</name>
 
-  <build>
-    <plugins>
-      <plugin>
-        <groupId>com.google.cloud.tools</groupId>
-        <artifactId>appengine-maven-plugin</artifactId>
-        <version>${appengine-maven-plugin.version}</version>
-        <configuration>
-          <version>1</version>
-          <project>travis-app-maven-plugin</project>
-          <version>GCLOUD_VERSION</version>
-          <jvmFlags>
-            <!-- required for kokoro gcp_ubuntu b/37217450 -->
-            <jvmFlag>-Djava.security.egd=file:/dev/urandom</jvmFlag>
-          </jvmFlags>
-          <environment>
-            <TEST_VAR>testVariableValue</TEST_VAR>
-          </environment>
-          <project>travis-app-maven-plugin</project>
-        </configuration>
-      </plugin>
-    </plugins>
-  </build>
 
   <profiles>
+    <profile>
+      <id>java8_and_below</id>
+      <activation>
+        <jdk>[,8]</jdk>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>com.google.cloud.tools</groupId>
+            <artifactId>appengine-maven-plugin</artifactId>
+            <version>${appengine-maven-plugin.version}</version>
+            <configuration>
+              <version>1</version>
+              <project>travis-app-maven-plugin</project>
+              <version>GCLOUD_VERSION</version>
+              <jvmFlags>
+                <!-- required for kokoro gcp_ubuntu b/37217450 -->
+                <jvmFlag>-Djava.security.egd=file:/dev/urandom</jvmFlag>
+              </jvmFlags>
+
+              <environment>
+                <TEST_VAR>testVariableValue</TEST_VAR>
+              </environment>
+              <project>travis-app-maven-plugin</project>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+
+    <profile>
+      <id>java9_and_up</id>
+      <activation>
+        <jdk>[9,]</jdk>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>com.google.cloud.tools</groupId>
+            <artifactId>appengine-maven-plugin</artifactId>
+            <version>${appengine-maven-plugin.version}</version>
+            <configuration>
+              <version>1</version>
+              <project>travis-app-maven-plugin</project>
+              <version>GCLOUD_VERSION</version>
+              <jvmFlags>
+                <!-- required for kokoro gcp_ubuntu b/37217450 -->
+                <jvmFlag>-Djava.security.egd=file:/dev/urandom</jvmFlag>
+
+                <!-- Illegal reflective access disallowed since Java 16 -->
```

Comment:
Out of curiosity, what do we get if we remove these `--add-opens` flags?

Reply:
```
java.lang.RuntimeException: Unable to create a DevAppServer
	at com.google.appengine.tools.development.DevAppServerFactory.doCreateDevAppServer(DevAppServerFactory.java:378)
...
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make static java.net.URLStreamHandler java.net.URL.getURLStreamHandler(java.lang.String) accessible: module java.base does not "opens java.net" to unnamed module @73a28541
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:199)
	at java.base/java.lang.reflect.Method.setAccessible(Method.java:193)
	at com.google.appengine.tools.development.StreamHandlerFactory.<clinit>(StreamHandlerFactory.java:52)

```

Reply:
Nice, I removed these flags and ran `mvn clean install` with java 17. It resulted in the following error:
```
: Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make static java.net.URLStreamHandler java.net.URL.getURLStreamHandler(java.lang.String) accessible: module java.base does not "opens java.net" to unnamed module @3ac3fd8b
testRunAsync: [INFO] GCLOUD: 	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
testRunAsync: [INFO] GCLOUD: 	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
testRunAsync: [INFO] GCLOUD: 	at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:199)
testRunAsync: [INFO] GCLOUD: 	at java.base/java.lang.reflect.Method.setAccessible(Method.java:193)
testRunAsync: [INFO] GCLOUD: 	at com.google.appengine.tools.development.StreamHandlerFactory.<clinit>(StreamHandlerFactory.java:52)
testRunAsync: [INFO] GCLOUD: 	... 12 more
```